substrate/frame/broker/src/migration.rs

```diff
@@ -156,6 +162,244 @@ mod v3 {
+		/// The length in relay chain blocks of the Leadin Period (where the price is decreasing).
+		pub leadin_length: BlockNumber,
+		/// The price of Bulk Coretime after the Leadin Period.
+		pub price: Balance,
+		/// The first timeslice of the Regions which are being sold in this sale.
+		pub region_begin: Timeslice,
+		/// The timeslice on which the Regions which are being sold in the sale terminate. (i.e.
+		/// One after the last timeslice which the Regions control.)
+		pub region_end: Timeslice,
+		/// The number of cores we want to sell, ideally. Selling this amount would result in no
+		/// change to the price for the next sale.
+		pub ideal_cores_sold: CoreIndex,
+		/// Number of cores which are/have been offered for sale.
+		pub cores_offered: CoreIndex,
+		/// The index of the first core which is for sale. Core of Regions which are sold have
+		/// incrementing indices from this.
+		pub first_core: CoreIndex,
+		/// The latest price at which Bulk Coretime was purchased until surpassing the ideal number
+		/// of cores were sold.
+		pub sellout_price: Option<Balance>,
+		/// Number of cores which have been sold; never more than cores_offered.
+		pub cores_sold: CoreIndex,
+	}
+}
+
+pub mod v4 {
+	use super::*;
+
+	type BlockNumberFor<T> = frame_system::pallet_prelude::BlockNumberFor<T>;
+
+	pub trait BlockToRelayHeightConversion<T: Config> {
+		/// Converts absolute value of parachain block number to relay chain block number
+		fn convert_block_number_to_relay_height(
+			block_number: BlockNumberFor<T>,
+		) -> RelayBlockNumberOf<T>;
+
+		/// Converts parachain block length into equivalent relay chain block length
+		fn convert_block_length_to_relay_length(
+			block_number: BlockNumberFor<T>,
+		) -> RelayBlockNumberOf<T>;
+	}
+
+	pub struct MigrateToV4Impl<T, BlockConversion>(PhantomData<T>, PhantomData<BlockConversion>);
+	impl<T: Config, BlockConversion: BlockToRelayHeightConversion<T>> UncheckedOnRuntimeUpgrade
+		for MigrateToV4Impl<T, BlockConversion>
+	{
+		#[cfg(feature = "try-runtime")]
+		fn pre_upgrade() -> Result<Vec<u8>, sp_runtime::TryRuntimeError> {
+			let (interlude_length, configuration_leadin_length) =
+				if let Some(config_record) = v3::Configuration::<T>::get() {
+					(config_record.interlude_length, config_record.leadin_length)
+				} else {
+					((0 as u32).into(), (0 as u32).into())
+				};
+
+			let updated_interlude_length: RelayBlockNumberOf<T> =
+				BlockConversion::convert_block_length_to_relay_length(interlude_length);
+			let updated_leadin_length: RelayBlockNumberOf<T> =
+				BlockConversion::convert_block_length_to_relay_length(configuration_leadin_length);
+			log::info!(target: LOG_TARGET, "Configuration Pre-Migration: Interlude Length {:?}->{:?} Leadin Length {:?}->{:?}", interlude_length, updated_interlude_length, configuration_leadin_length, updated_leadin_length);
+
+			let (sale_start, sale_info_leadin_length) =
+				if let Some(sale_info_record) = v3::SaleInfo::<T>::get() {
```

Comment:
In `#[try-runtime]` feature, you might as well feel free to panic.

Reply:
SaleInfo might not exist if `start_sales` pallet call is not called yet